crates/interfaces/src/p2p/headers/downloader.rs

```diff
@@ -12,7 +13,9 @@ use reth_primitives::{BlockHashOrNumber, SealedHeader, H256};
 /// of fulfilling these requests.
 ///
 /// A [HeaderDownloader] is a [Stream] that returns batches of headers.
-pub trait HeaderDownloader: Send + Sync + Stream<Item = Vec<SealedHeader>> + Unpin {
+pub trait HeaderDownloader:
+    Send + Sync + Stream<Item = HeadersDownloaderResult<Vec<SealedHeader>>> + Unpin
+{
```

Comment:
making this a result is a good call, this gets rid of the underlying design flaw where the downloader is expected to always close the gap eventually